docs/changelog.rst

```diff
@@ -16,8 +16,146 @@ IN DEVELOPMENT
 Changes will be summarized here periodically.
 
 
+New Features:
+
+* ``pybind11/gil_safe_call_once.h`` was added (it needs to be included
+  explicitly). The primary use case is GIL-safe initialization of C++
+  ``static`` variables.
+  `#4877 <https://github.com/pybind/pybind11/pull/4877>`_
+
+* Support move-only iterators in ``py::make_iterator``,
+  ``py::make_key_iterator``, ``py::make_value_iterator``.
+  `#4834 <https://github.com/pybind/pybind11/pull/4834>`_
+
+* Two simple ``py::set_error()`` functions were added and the documentation was
+  updated accordingly. In particular, ``py::exception<>::operator()`` was
+  deprecated (use one of the new function instead). The documentation for
+  ``py::exception<>`` was further updated to not suggest code that may result
+  in undefined behavior.
+  `#4772 <https://github.com/pybind/pybind11/pull/4772>`_
+
+Bug fixes:
+
+* ``pybind11`` now supports compiling for NumPy 2.  The required changes are
```

Comment:
Move this up to New Features section, very top?

Suggested replacement of text:

`pybind11` now support compiling for NumPy 2. Note that NumPy 2 introduces breaking changes (in general, not related to pybind11). Please see https://numpy.org/devdocs/numpy_2_0_migration_guide.html in case of unexpected behavior. If you experience issues, you may want to experiment with the `PYBIND11_NUMPY_1_ONLY` define (ideally set from the command line!) to disable the NumPy 2 support in pybind11. However, this define may be removed in the future, because it can be a source of ODR violations if used inconsistently.

________

I really wouldn't want to play down the changes ("small and should not matter"). I expect a significant amount of friction related to the NumPy 2 migration. Better not to give people a false sense of safety.

Reply:
I meant to put something in the migration guide, I'll do that too.

Reply:
Note that "NumPy 2 introduces breaking changes (in general, not related to pybind11)" - I'm not sure this is needed, it wouldn't be called NumPy 2 if it had no breaking changes. I think we can still link to it without explicitly stating the obvious. Also, this define will be removed in the future because 2.0 replaces 1.x and we won't maintain a compat option for a dead package forever (also, anyway).

Reply:
Reworked.